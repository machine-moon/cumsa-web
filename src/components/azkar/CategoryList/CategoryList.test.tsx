import React from "react";
import { render } from "@testing-library/react";
import { CategoryList } from "./index";

// Mock the AudioPlayer component
jest.mock("@/components/azkar/AudioPlayer", () => ({
  AudioPlayer: ({ id, isActive, loading }: { id: number; isActive: boolean; loading: boolean }) => (
    <div data-testid="audio-player" data-id={id} data-active={isActive} data-loading={loading}>
      {loading ? "Loading..." : isActive ? "Pause" : "Play"}
    </div>
  ),
}));

describe("CategoryList", () => {
  const mockCategories = [
    { ID: 1, TITLE: "Morning Azkar", AUDIO_URL: "", TEXT: "https://example.com/morning.json" },
    {
      ID: 2,
      TITLE: "Evening Azkar",
      AUDIO_URL: "https://example.com/evening.mp3",
      TEXT: "https://example.com/evening.json",
    },
    { ID: 3, TITLE: "Sleep Azkar", AUDIO_URL: "", TEXT: "https://example.com/sleep.json" },
  ];

  const defaultProps = {
    categories: mockCategories,
    onSelect: jest.fn(),
    onPlayAudio: jest.fn(),
    isPlaying: jest.fn(() => false),
    audioLoading: false,
    audioError: null,
  };

  it("renders correctly with categories", () => {
    const { container } = render(<CategoryList {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with active category", () => {
    const { container } = render(<CategoryList {...defaultProps} activeId={1} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with audio loading", () => {
    const { container } = render(
      <CategoryList {...defaultProps} audioLoading={true} isPlaying={() => true} />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with audio error", () => {
    const { container } = render(
      <CategoryList {...defaultProps} audioError="Failed to load audio" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with invalid categories", () => {
    const { container } = render(<CategoryList {...defaultProps} categories={null} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with empty categories", () => {
    const { container } = render(<CategoryList {...defaultProps} categories={[]} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
