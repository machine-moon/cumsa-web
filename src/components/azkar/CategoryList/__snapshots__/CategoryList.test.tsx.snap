// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList renders correctly with active category 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-[var(--blue)]/20 overflow-y-auto h-full bg-gradient-to-b from-white via-[#f1fdff] to-[#e3f7fa] text-xs sm:text-sm custom-scroll"
>
  <ul
    class="p-2 space-y-1"
  >
    <li
      class="rounded transition-colors animate-fade-in bg-[var(--blue)]/15 ring-1 ring-[var(--blue)]/40"
      style="animation-delay: 0ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--blue)]"
        >
          Morning Azkar
        </span>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 40ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Evening Azkar
        </span>
        <div
          class="flex justify-center pt-1"
        >
          <div
            data-active="false"
            data-id="cat-2"
            data-loading="false"
            data-testid="audio-player"
          >
            Play
          </div>
        </div>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 80ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Sleep Azkar
        </span>
      </div>
    </li>
  </ul>
</aside>
`;

exports[`CategoryList renders correctly with audio error 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-[var(--blue)]/20 overflow-y-auto h-full bg-gradient-to-b from-white via-[#f1fdff] to-[#e3f7fa] text-xs sm:text-sm custom-scroll"
>
  <ul
    class="p-2 space-y-1"
  >
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 0ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Morning Azkar
        </span>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 40ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Evening Azkar
        </span>
        <div
          class="flex justify-center pt-1"
        >
          <div
            data-active="false"
            data-id="cat-2"
            data-loading="false"
            data-testid="audio-player"
          >
            Play
          </div>
        </div>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 80ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Sleep Azkar
        </span>
      </div>
    </li>
  </ul>
</aside>
`;

exports[`CategoryList renders correctly with audio loading 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-[var(--blue)]/20 overflow-y-auto h-full bg-gradient-to-b from-white via-[#f1fdff] to-[#e3f7fa] text-xs sm:text-sm custom-scroll"
>
  <ul
    class="p-2 space-y-1"
  >
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 0ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Morning Azkar
        </span>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 40ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Evening Azkar
        </span>
        <div
          class="flex justify-center pt-1"
        >
          <div
            data-active="true"
            data-id="cat-2"
            data-loading="true"
            data-testid="audio-player"
          >
            Loading...
          </div>
        </div>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 80ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Sleep Azkar
        </span>
      </div>
    </li>
  </ul>
</aside>
`;

exports[`CategoryList renders correctly with categories 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-[var(--blue)]/20 overflow-y-auto h-full bg-gradient-to-b from-white via-[#f1fdff] to-[#e3f7fa] text-xs sm:text-sm custom-scroll"
>
  <ul
    class="p-2 space-y-1"
  >
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 0ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Morning Azkar
        </span>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 40ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Evening Azkar
        </span>
        <div
          class="flex justify-center pt-1"
        >
          <div
            data-active="false"
            data-id="cat-2"
            data-loading="false"
            data-testid="audio-player"
          >
            Play
          </div>
        </div>
      </div>
    </li>
    <li
      class="rounded transition-colors animate-fade-in hover:bg-[var(--blue)]/8"
      style="animation-delay: 80ms;"
    >
      <div
        class="w-full px-3 py-2 flex flex-col gap-2 cursor-pointer"
        role="button"
        tabindex="0"
      >
        <span
          class="text-sm font-medium text-[var(--foreground)]"
        >
          Sleep Azkar
        </span>
      </div>
    </li>
  </ul>
</aside>
`;

exports[`CategoryList renders correctly with empty categories 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-[var(--blue)]/20 overflow-y-auto h-full bg-gradient-to-b from-white via-[#f1fdff] to-[#e3f7fa] text-xs sm:text-sm custom-scroll"
>
  <ul
    class="p-2 space-y-1"
  />
</aside>
`;

exports[`CategoryList renders correctly with invalid categories 1`] = `
<aside
  class="w-60 sm:w-64 shrink-0 border-r border-black/10 h-full flex items-center justify-center text-[10px] text-[var(--red)]"
>
  Invalid categories
</aside>
`;
